ingest/src/test/java/com/connexta/ingest/transform/TransformClientTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.ingest.transform;
+
+import static com.connexta.ingest.transform.ExtendedMockRestResponseCreators.withBadRequest;
+import static com.connexta.ingest.transform.ExtendedMockRestResponseCreators.withServerError;
+import static com.connexta.ingest.transform.ExtendedMockRestResponseCreators.withSuccess;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.anything;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.jsonPath;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.method;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.net.URISyntaxException;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class TransformClientTest {
+
+  @Autowired RestTemplate restTemplate;
+
+  @Autowired TransformClient client;
+
+  private MockRestServiceServer mockServer;
+  private String transformEndpoint;
+  private TransformResponse expectedSuccessResponse;
+  private TransformRequest transformRequest;
+  private TransformResponse expectedFailedResponse;
+
+  @Before
+  public void setUp() throws URISyntaxException {
+    mockServer = MockRestServiceServer.createServer(restTemplate);
+    transformEndpoint = "http://localhost:8000/transform";
```

Comment:
Is this url just a test one that you don't actually use while the test is running?
If you are actually using the url in the test (which I'm not 100% clear on), there's a way to get the actual running container host and port using spring.

Reply:
It's a test that should use actual HTTP requests and responses, but the mock server will actually service the requests. The mock server only exists for the lifetime of the test. It is not meant to invoke services provided by another container.